syntax/text/parser2/src/Luna/Syntax/Text/Parser/Ast/Spanned.hs

```diff
@@ -0,0 +1,360 @@
+{-# INLINE apps #-}
+
+missing :: Spanned Ast
+missing = Spanned mempty Missing
+{-# INLINE missing #-}
+
+list :: [Spanned Ast] -> Spanned Ast
+list = inheritSpanList $ \items -> List items
+{-# INLINE list #-}
+
+unit :: Spanned Ast -> Spanned Ast
+unit = inheritSpan1 $ \items -> Unit items
+{-# INLINE unit #-}
+
+block :: NonEmpty (Spanned Ast) -> Spanned Ast
+block = inheritSpanList1 $ \items -> Block items
+{-# INLINE block #-}
+
+isOperator :: Ast -> Bool
+isOperator = \case
+    AstOperator {} -> True
+    _ -> False
+{-# INLINE isOperator #-}
+
+documented :: Spanned Ast -> Spanned Ast -> Spanned Ast
+documented = inheritSpan2 $ \doc base -> Documented doc base
+{-# INLINE documented #-}
+
+
+
+-----------------------------
+-- === Unwrapping span === --
+-----------------------------
+
+-- | See docs of unspan.
+unsafeUnspan :: Spanned a -> a
+unsafeUnspan = \(Spanned _ a) -> a
+{-# INLINE unsafeUnspan #-}
+
+-- | Removing span is not always secure as it causes the child spans to be
+--   incorrect if used with different parrent. Thus the secure way to remove
+--   span is to embed the parent offset span into first child offset span while
+--   unwrapping.
+unspan :: Spanned Ast -> Ast
+unspan = \(Spanned cs a) ->
+    let span = CodeSpan.dropLength cs
+    in case a of
+        AstVar          a -> AstVar         $ prependSpan span a
+        AstCons         a -> AstCons        $ prependSpan span a
+        AstOperator     a -> AstOperator    $ prependSpan span a
+        AstModifier     a -> AstModifier    $ prependSpan span a
+        AstWildcard     a -> AstWildcard    $ prependSpan span a
+        AstNumber       a -> AstNumber      $ prependSpan span a
+        AstStr          a -> AstStr         $ prependSpan span a
+        AstBlock        a -> AstBlock       $ prependSpan span a
+        AstMarker       a -> AstMarker      $ prependSpan span a
+        AstLineBreak    a -> AstLineBreak   $ prependSpan span a
+        AstComment      a -> AstComment     $ prependSpan span a
+        AstDocumented   a -> AstDocumented  $ prependSpan span a
+        AstMetadata     a -> AstMetadata    $ prependSpan span a
+        AstInvalid      a -> AstInvalid     $ prependSpan span a
+        AstApp          a -> AstApp         $ prependSpan span a
+        AstInfixApp     a -> AstInfixApp    $ prependSpan span a
+        AstMissing      a -> AstMissing     $ prependSpan span a
+        AstList         a -> AstList        $ prependSpan span a
+        AstUnit         a -> AstUnit        $ prependSpan span a
+        AstSectionLeft  a -> AstSectionLeft $ prependSpan span a
+        AstSectionRight a -> AstSectionRight$ prependSpan span a
+{-# INLINE unspan #-}
+
+
+
+-------------------------
+-- === PrependSpan === --
+-------------------------
+
+-- === Definition === --
+
+class PrependSpan a where
+    prependSpan :: CodeSpan -> a -> a
+    prependSpan = \_ -> id
+
+
+-- === Utils === --
+
+prepSpan :: CodeSpan -> (Spanned b -> Spanned b)
```

Comment:
Is this 'prepare' or 'prepend'? If it's the latter it shouldn't be abbreviated to 'prep' as that is an actual word.